File: 1000.py
def diff(a,b):
    n=0
    for i in range(len(a)):
        if a[i] != b[i]:
            n+=1
            if n > 2 :
                return True
    return False

def main():
    data = []
    l = int(input())
    n = int(input())
    for i in range(n):
        data.append(input())
    for i in range(0,n-1):
        if diff(data[i],data[i+1]):
            print(data[i])
            return 0
    print(data[-1])

if __name__ == '__main__':
    main()
    
'''
4
12
HEAD
HEAP
LEAP
TEAR
REAR
BAER
BAET
BEEP
JEEP
JOIP
JEIP
AEIO

REAR
'''

'''
4
5
HEAD
HEAP
LEAP
TEAR
REAR
'''

File: c++/Programming/10/1000.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){
    int L, N;
    cin >> L >> N;
    vector<string> words(N);
    cin >> words[0];
    for(int i = 1; i < N; i++){
        cin >> words[i];
    }

    for(int i = 1; i < N; i++){
        int cnt = 0;
        for(int j = 0; j < L; j++){
            if(words[i][j] != words[i - 1][j]){
                cnt++;
            }
            if(cnt > 2){
                cout << words[i - 1];
                return 0;
            }
        }
    }
    cout << words[N - 1];
    return 0;
}

/*
4
12
HEAD
HEAP
LEAP
TEAR
REAR
BAER
BAET
BEEP
JEEP
JOIP
JEIP
AEIO
*/